src/weather_man.js

```diff
@@ -9,22 +9,45 @@ class WeatherMan extends React.Component {
     this.state = {
       city: "",
       weather: "",
+      message: "",
     }
+    this.registerCity = this.registerCity.bind(this);
+    this.updateMessage = this.updateMessage.bind(this);
   }
 
-  componentDidMount() {
-    getDataFor(this.props.city)
+  registerCity(event) {
+    this.setState({
+      city: event.target.value,
+    })
+  }
+
+  registerData() {
+    getDataFor(this.state.city)
       .then(data => {
 	this.setState({
 	  city: data.name,
 	  weather: data.weather[0].description,
-	});
+	})
       });
   }
 
+  updateMessage() {
+    this.registerData();
+    this.setState({
```

Comment:
Seems like a possible race condition here, as `registerData()` will call `setState()` once the promise is resolved and this function calls `setState()` directly after. And since `getDataFor()` runs asynchronously I think it's possible that the `message` will be updated before the `fetch` call returns.

You could try and do both things in one function. Like inlining `registerData()` so that you can set the state for `city`, `weather` as well as `message` in one go.